File: algorithms/c#/CountingSort.cs
using System.Linq;

namespace Algorithms
{
	public static class CountingSort
	{
		public static void Sort(int[] array)
		{
			int min = array.Min();
			int max = array.Max();
			int range = (max - min) + 1;
			int[] count = new int[range];
			int[] sortedArray = new int[array.Length];

			for(int i = 0; i < array.Length; i++) count[array[i] - min]++;
			for(int i = 1; i < count.Length; i++) count[i] += count[i - 1];

			for(int i = array.Length - 1; i >= 0; i--)
			{
				sortedArray[count[array[i] - min] - 1] = array[i];
				count[array[i] - min]--;
			}

			for(int i = 0; i < array.Length; i++)
			{
				array[i] = sortedArray[i];
			}
		}
	}
}

File: algorithms/javascript/counting-sort.js
function countingSort(array)
{
	let min = Math.min(...array);
	let max = Math.max(...array);
	let range = (max - min) + 1;
	let count = new Array(range).fill(0);
	let sortedArray = new Array(array.length).fill(0);

	for(let i = 0; i < array.length; i++) count[array[i] - min]++;
	for(let i = 1; i < count.length; i++) count[i] += count[i - 1];

	for(let i = array.length - 1; i >= 0; i--)
	{
		sortedArray[count[array[i] - min] - 1] = array[i];
		count[array[i] - min]--;
	}

	for(let i = 0; i < array.length; i++)
	{
		array[i] = sortedArray[i];
	}

	return sortedArray;
}